### PR TITLE
Incremental unit test for MDRange with views uptil 4 ranks.

### PR DESCRIPTION
**Pull request by rgayatri23**
Similar to MDRange incremental test without views, except here we use views and lambdas.

### CODE REVIEW

File: core/unit_test/incremental/Test05_MDRangePolicy.hpp [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+  // An MDRangePolicy for 4 nested loops
+  using MDPolicyType_4D = typename Kokkos::Experimental::MDRangePolicy<
+      ExecSpace, Kokkos::Experimental::Rank<4>, int_index>;
+
+  // compare and equal
+  void compare_equal_2D() {
+    for (int i = 0; i < N; ++i)
+      for (int j = 0; j < M; ++j)
+        ASSERT_EQ(hostDataView_2D(i, j), i * j * delta);
```
Comment (crtrott) [inline core/unit_test/incremental/Test05_MDRangePolicy.hpp]: lets fill this with something which is unique per pair `(i*M+j)` for example

Comment (rgayatri23) [reply core/unit_test/incremental/Test05_MDRangePolicy.hpp]: Well i*j*delta is unique per (i,j) pair.

Comment (mhoemmen) [reply core/unit_test/incremental/Test05_MDRangePolicy.hpp]: Christian is pointing out that `i * j * delta` equals `j * i * delta`, so you can't tell the difference between the View and its transpose.

Comment (janciesko) [reply core/unit_test/incremental/Test05_MDRangePolicy.hpp]: i * N + j might be what we want here. Let's scratch the scale factor (delta)

Comment (crtrott) [reply core/unit_test/incremental/Test05_MDRangePolicy.hpp]: jup exactly.

Comment (rgayatri23) [reply core/unit_test/incremental/Test05_MDRangePolicy.hpp]: Ah yes, ok will update that.

Comment (rgayatri23) [reply core/unit_test/incremental/Test05_MDRangePolicy.hpp]: Retest this please.

**Pull request merged**